src/main/java/com/yscope/logging/log4j1/AbstractBufferedRollingFileAppender.java

```diff
@@ -504,11 +513,13 @@ private class SyncRequest extends Request {
       public final String logFileBaseName;
       public final long logRolloverTimestamp;
       public final boolean deleteFile;
+      public final Map<String, Object> metadata;
 
       public SyncRequest (String logFileBaseName, long logRolloverTimestamp, boolean deleteFile) {
         this.logFileBaseName = logFileBaseName;
         this.logRolloverTimestamp = logRolloverTimestamp;
         this.deleteFile = deleteFile;
+        this.metadata = computeSyncRequestMetadata();
```

Comment:
This seems a bit messy:
* We're depending on SyncRequest being constructed from the parent class here whereas that might not always be the case in which case the request metadata may be for a different file than the one that the sync request is for.
* It breaks the convention we've already established where all the members are passed in.

I would suggest simply passing in the metadata as a parameter and adding a call to computeSyncRequestMetadata at every callsite of addSyncRequest.